File: ios/Pantheon/Models/HorusModels.swift
import Foundation

// MARK: - Horus (Code Graph)

struct HorusSymbol: Codable, Identifiable {
    var id: String { "\(file):\(name):\(line)" }

    let name: String
    let kind: String
    let file: String
    let line: Int
    let endLine: Int
    let signature: String
    let doc: String?
    let exported: Bool
    let parent: String?

    var kindIcon: String {
        switch kind {
        case "func":      return "f()"
        case "method":    return "m()"
        case "type":      return "T"
        case "struct":    return "S"
        case "interface": return "I"
        case "const":     return "C"
        case "var":       return "V"
        case "package":   return "P"
        default:          return "?"
        }
    }
}

struct HorusSymbolGraph: Codable {
    let root: String
    let packages: [String]
    let symbols: [HorusSymbol]
    let stats: HorusGraphStats
    let builtAt: String
}

struct HorusGraphStats: Codable {
    let files: Int
    let packages: Int
    let types: Int
    let functions: Int
    let methods: Int
    let interfaces: Int
    let totalLines: Int
}

struct HorusOutlineResult: Codable {
    let outline: String
}

struct HorusContextResult: Codable {
    let context: String
}

File: android/app/src/main/java/ai/sirsi/pantheon/models/HorusModels.kt
package ai.sirsi.pantheon.models

import kotlinx.serialization.Serializable

// --- Horus (Code Graph) ---

@Serializable
data class HorusSymbol(
    val name: String,
    val kind: String,
    val file: String,
    val line: Int,
    val endLine: Int = 0,
    val signature: String = "",
    val doc: String? = null,
    val exported: Boolean = false,
    val parent: String? = null,
) {
    val id: String get() = "$file:$name:$line"

    val kindIcon: String
        get() = when (kind) {
            "func"      -> "f()"
            "method"    -> "m()"
            "type"      -> "T"
            "struct"    -> "S"
            "interface" -> "I"
            "const"     -> "C"
            "var"       -> "V"
            "package"   -> "P"
            else        -> "?"
        }

    val displayKind: String
        get() = when (kind) {
            "func"      -> "Function"
            "method"    -> "Method"
            "type"      -> "Type"
            "struct"    -> "Struct"
            "interface" -> "Interface"
            "const"     -> "Constant"
            "var"       -> "Variable"
            "package"   -> "Package"
            else        -> kind
        }
}

@Serializable
data class HorusGraphStats(
    val files: Int,
    val packages: Int,
    val types: Int,
    val functions: Int,
    val methods: Int,
    val interfaces: Int,
    val totalLines: Int,
)

@Serializable
data class HorusSymbolGraph(
    val root: String,
    val packages: List<String>,
    val symbols: List<HorusSymbol>,
    val stats: HorusGraphStats,
    val builtAt: String,
)

@Serializable
data class HorusOutlineResult(
    val outline: String,
)

@Serializable
data class HorusContextResult(
    val context: String,
)
